packages/wdio-allure-reporter/src/constants.ts

```diff
@@ -1,43 +1,47 @@
-const PASSED = 'passed'
-const FAILED = 'failed'
-const BROKEN = 'broken'
-const PENDING = 'pending'
-const CANCELED = 'canceled'
-const SKIPPED = 'skipped'
-
-const testStatuses = {
-    PASSED,
-    FAILED,
-    BROKEN,
-    PENDING
-}
-
-const stepStatuses = {
-    PASSED,
-    FAILED,
-    BROKEN,
-    CANCELED,
-    SKIPPED
-}
-
-const events = {
-    addLabel: 'allure:addLabel',
-    addFeature: 'allure:addFeature',
-    addStory: 'allure:addStory',
-    addSeverity: 'allure:addSeverity',
-    addIssue: 'allure:addIssue',
-    addTestId: 'allure:addTestId',
-    addEnvironment: 'allure:addEnvironment',
-    addDescription: 'allure:addDescription',
-    addAttachment: 'allure:addAttachment',
-    startStep: 'allure:startStep',
-    endStep: 'allure:endStep',
-    addStep: 'allure:addStep',
-    addArgument: 'allure:addArgument'
-}
-
-const mochaEachHooks = ['"before each" hook', '"after each" hook']
-const mochaAllHooks = ['"before all" hook', '"after all" hook']
-const linkPlaceholder = '{}'
-
-export { testStatuses, stepStatuses, events, mochaEachHooks, mochaAllHooks, linkPlaceholder }
+import Allure from 'allure-js-commons'
+
+export const PASSED = 'passed'
+export const FAILED = 'failed'
+export const BROKEN = 'broken'
+export const PENDING = 'pending'
+export const CANCELED = 'canceled'
+export const SKIPPED = 'skipped'
+
+const testStatuses: Record<string, Allure.Status> = {
+    PASSED,
+    FAILED,
+    BROKEN,
+    PENDING,
+    CANCELED
+}
+const stepStatuses: Record<string, Allure.Status> = {
+    PASSED,
+    FAILED,
+    BROKEN,
+    PENDING,
+    CANCELED,
+    SKIPPED
+}
+
+// Node 12 types only accept pre-defined strings
+const events: any = {

```

Comment:
```suggestion
const events: Record<string, string> = {

```

Reply:
This does not work because Node 12 hard coded the event string for `process.on()`. Casting to any is the only way now.

Reply:
I would prefer to cast it to any in the `process.on` call. This way we can ensure type safety here and just ignore it there.